lib/events/filelog.go

```diff
@@ -222,14 +223,124 @@ func (l *FileLog) SearchUnstructuredEvents(ctx context.Context, req SearchEvents
 	return evts, next, nil
 }
 
-func (l *FileLog) searchEventsWithFilter(fromUTC, toUTC time.Time, limit int, order types.EventOrder, startAfter string, filter searchEventsFilter) ([]EventFields, string, error) {
+// checkpointKey is a pointer to an event in the log. This is used to keep track
+// of the last processed event so the caller can resume querying from this event.
+type checkpointKey struct {
+	id        string
+	timestamp time.Time
+}
+
+// MarshalText marshals a checkpointKey into plain text.
+func (c checkpointKey) MarshalText() ([]byte, error) {
+	return []byte(c.String()), nil
+}
+
+// UnmarshalText unmarshals a checkpointKey from its text representation.
+func (c *checkpointKey) UnmarshalText(b []byte) error {
+	if len(b) == 0 {
+		return nil
+	}
+	id, rawTime, ok := strings.Cut(string(b), "/")
```

Comment:
nit: hold on to a shorter piece of string for `id` since we're going to throw away the time part

```suggestion
	id, rawTime, ok := bytes.Cut(b, "/")
```

(and then convert the separate parts to strings)